SocialCareCaseViewerApi/V1/Gateways/DatabaseGateway.cs

```diff
@@ -480,6 +480,356 @@ public void UpdatePerson(UpdatePersonRequest request)
             _databaseContext.SaveChanges();
         }
 
+        public void PatchPerson(PatchPersonRequest request)
+        {
+            Person person = _databaseContext
+                .Persons
+                .Include(x => x.Addresses)
+                .Include(x => x.PhoneNumbers)
+                .Include(x => x.OtherNames)
+                .Include(x => x.KeyContacts)
+                .Include(x => x.GpDetails)
+                .Include(x => x.TechUse)
+                .Include(x => x.Disability)
+                .Include(x => x.Emails)
+                .Include(x => x.LastUpdated)
+                .FirstOrDefault(x => x.Id == request.Id);
+
+            if (person == null)
+            {
+                throw new UpdatePersonException("Person not found");
+            }
+
+            if (request.ContextFlag != null)
+            {
+                person.AgeContext = request.ContextFlag;
+            }
+            if (request.DateOfBirth != null)
+            {
+                person.DateOfBirth = request.DateOfBirth;
+            }
+            if (request.DateOfDeath != null)
+            {
+                person.DateOfDeath = request.DateOfDeath;
+            }
+            if (request.EmailAddress != null)
+            {
+                person.EmailAddress = request.EmailAddress;
+            }
+            if (request.EmailAddress != null)
+            {
+                person.EmailAddress = request.EmailAddress;
+            }
+            if (request.Ethnicity != null)
+            {
+                person.Ethnicity = request.Ethnicity;
+            }
+            if (request.FirstLanguage != null)
+            {
+                person.FirstLanguage = request.FirstLanguage;
+            }
+            if (request.FirstName != null)
```

Comment:
`person.FirstLanguage = request.FirstLanguage ?? person.FirstLanguage;` might make these easier to read.

Reply:
Refactored